src/tree/service.rs

```diff
@@ -95,92 +59,54 @@ impl<M: Middleware> TreeAvailabilityService<M> {
             axum::Server::bind(&addr)
                 .serve(router.into_make_service())
                 .await
-                .map_err(TreeAvailabilityError::HyperError)?;
-            tracing::info!("Server spawned");
-
-            Ok(())
+                .expect("axum server failed");
         });
 
         handles.push(server_handle);
 
-        // Spawn a new task to keep the world tree synced to the chain head
-        handles.push(self.world_tree.spawn());
-
-        handles
+        Ok(handles)
     }
 }
 
 #[derive(Serialize, Deserialize, Debug)]
 #[serde(rename_all = "camelCase", deny_unknown_fields)]
 pub struct InclusionProofRequest {
     pub identity_commitment: Hash,
-    pub root: Option<Hash>,
+    chain_id: Option<ChainId>,
 }
 
 impl InclusionProofRequest {
     pub fn new(
         identity_commitment: Hash,
-        root: Option<Hash>,
+        chain_id: Option<ChainId>,
     ) -> InclusionProofRequest {
         Self {
             identity_commitment,
-            root,
+            chain_id,
         }
     }
 }
 
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+pub struct ChainIdQueryParams {
+    chain_id: Option<ChainId>,
+}
+
 #[tracing::instrument(level = "debug", skip(world_tree))]
-pub async fn inclusion_proof<M: Middleware>(
+pub async fn inclusion_proof<M: Middleware + 'static>(
     State(world_tree): State<Arc<WorldTree<M>>>,
+    Query(query_params): Query<ChainIdQueryParams>,
     Json(req): Json<InclusionProofRequest>,
 ) -> Result<(StatusCode, Json<Option<InclusionProof>>), TreeError> {
-    if world_tree.synced.load(Ordering::Relaxed) {
-        let inclusion_proof = world_tree
-            .tree_data
-            .read()
-            .await
-            .get_inclusion_proof(req.identity_commitment, req.root);
-
-        Ok((StatusCode::OK, inclusion_proof.into()))
-    } else {
-        Err(TreeError::TreeNotSynced)
-    }
-}
-
-#[derive(Debug, Serialize, Deserialize)]
-#[serde(rename_all = "camelCase")]
-pub struct SyncResponse {
-    pub synced: bool,
-    pub block_number: Option<u64>,
-}
+    let chain_id = query_params.chain_id.or(req.chain_id);
```

Comment:
It's kinda weird that query params chain id overrides the one from the request. Maybe let's ditch the request one?